pointcloud/pointcloud_file.go

```diff
@@ -529,12 +544,33 @@ func readPCDHelper(inRaw io.Reader, pctype PCType) (PointCloud, error) {
 		pc = NewWithPrealloc(int(header.points))
 	case KDTreeType:
 		pc = NewKDTreeWithPrealloc(int(header.points))
+	case BasicOctreeType:
+		var meta MetaData
+		switch header.data {
+		case PCDAscii:
+			meta, err = getPCDMetaDataASCII(*in, header)
+		case PCDBinary:
+			meta, err = getPCDMetaDataBinary(*in, header)
+		case PCDCompressed:
+			// return readPCDCompressed(in, header)
+			return nil, errors.New("compressed pcd not yet supported")
+		default:
+			return nil, fmt.Errorf("unsupported pcd data type %v", header.data)
+		}
```

Comment:
It feels like this switch statement could be within the function more general called `getPCDMetaData`